File: Example_Clients/CameraClient/MicrosoftCognitiveServiceExample/UserInterface.cs
using System.Drawing;
using System.Windows.Forms;
using System;
using System.IO;

namespace MicrosoftCognitiveServiceExample
{
    public class UserInterface : Form
    {
        public static string DEFAULT_IP;
        public static string SUBSCRIPTION_KEY_FOR_EMOTION;
        public static string SUBSCRIPTION_KEY_FOR_VISION;
        public static string PICTURE_PATH;

        public static MicrosoftCognitiveServiceRequests MicrosoftCognitiveServiceRequests = new MicrosoftCognitiveServiceRequests();        
        public static Button AnalyzeButton;
        private Button ConnectButton;
        public static Button AutomaticButton;
        public static bool AutomaticActive = false;
        public static TextBox CameraUrl;
        private ComboBox CameraDriverComboBox;
        public static TextBox SubsciptionKey;
        public static Panel CameoPanel;
        public static PictureBox CameraSnapshot;
        public static Label ResultsOfAnalysis;
        private CameraConnector CameraConnector;        

        public UserInterface()
        {
            string ConfigFileName = "..\\..\\..\\Resources\\MSCS.cfg";
            try
            {                
                Configuration.ReadConfig(ConfigFileName);
            }
            catch (Exception ex)
            {
               Console.Out.WriteLine(ex.StackTrace);

               Console.Out.WriteLine("Configfile " + ConfigFileName + " not found");
                
               Console.Out.WriteLine("Please define a configfile named MSCS.cfg to prefill Camera IP and Subscription Key");
            }

            this.InitializeGUIComponents();

            CameraConnector = new CameraConnector(ConnectButton, CameoPanel);
        }

        private void InitializeGUIComponents()
        {
            // 
            // URL Label
            // 
            Label URLLabel = new System.Windows.Forms.Label();
            URLLabel.Text = "Camera IP : ";
            URLLabel.Size = new System.Drawing.Size(150, 20);
            URLLabel.Font = new System.Drawing.Font("Arial", 14F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            URLLabel.Location = new System.Drawing.Point(0, 0);
            URLLabel.Name = "Camera URL Label";

            // 
            // URL Textbox
            // 
            CameraUrl = new System.Windows.Forms.TextBox();
            CameraUrl.Text = DEFAULT_IP;
            CameraUrl.Size = new System.Drawing.Size(150, 20);
            CameraUrl.Font = new System.Drawing.Font("Arial", 14F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            CameraUrl.Location = new System.Drawing.Point(150, 0);
            CameraUrl.Name = "Camera URL Textbox";

            // 
            // Camera Driver Label
            // 
            Label CameraDriverLabel = new System.Windows.Forms.Label();
            CameraDriverLabel.Text = "Camera Driver : ";
            CameraDriverLabel.Size = new System.Drawing.Size(150, 30);
            CameraDriverLabel.Font = new System.Drawing.Font("Arial", 14F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            CameraDriverLabel.Location = new System.Drawing.Point(450, 0);
            CameraDriverLabel.Name = "Camera Driver Label";

            // 
            // ComboBoxProgID
            // 
            CameraDriverComboBox = new System.Windows.Forms.ComboBox();
            CameraDriverComboBox.Name = "Camera Driver";
            CameraDriverComboBox.Items.AddRange(new object[] {
            "GCA.VIP.DeviceProxy",
            "GCA.ONVIF.DeviceProxy",
            "GCA.RTSP.DeviceProxy"});
            CameraDriverComboBox.Location = new System.Drawing.Point(600, 0);
            CameraDriverComboBox.Font = new System.Drawing.Font("Arial", 14F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            CameraDriverComboBox.Size = new System.Drawing.Size(250, 30);
            CameraDriverComboBox.SelectedIndex = 0;


            // 
            // Subscribtion Key Label
            // 
            Label SubsciptionKeyLabel = new System.Windows.Forms.Label();
            SubsciptionKeyLabel.Text = "Subscribtion Key : ";
            SubsciptionKeyLabel.Size = new System.Drawing.Size(200, 20);
            SubsciptionKeyLabel.Font = new System.Drawing.Font("Arial", 14F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            SubsciptionKeyLabel.Location = new System.Drawing.Point(1250, 0);
            SubsciptionKeyLabel.Name = "Subscribtion Key Label";

            // 
            // Subscribtion Key Textbox
            // 
            SubsciptionKey = new System.Windows.Forms.TextBox();
            SubsciptionKey.Text = SUBSCRIPTION_KEY_FOR_EMOTION;
            SubsciptionKey.Size = new System.Drawing.Size(400, 20);
            SubsciptionKey.Font = new System.Drawing.Font("Arial", 14F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            SubsciptionKey.Location = new System.Drawing.Point(1450, 0);
            SubsciptionKey.Name = "Subscribtion Key Textbox";

            // 
            // CameraSnapshotBox
            //         
            CameraSnapshot = new System.Windows.Forms.PictureBox();

            // Construct an image object from a file in the local directory.
            // ... This file must exist in the solution.
            string ImageFileName = "..\\..\\..\\Resources\\SmartEdgeDevices.png"; 
            Image image = new Bitmap(Image.FromFile(ImageFileName), new Size(800, 450));
            // Set the PictureBox image property to this image.
            // ... Then, adjust its height and width properties.
            if (image != null)
            {
                CameraSnapshot.Image = image;
                CameraSnapshot.Location = new System.Drawing.Point(800, 30);
                CameraSnapshot.Name = "CameraSnapshotBox";
                CameraSnapshot.SizeMode = PictureBoxSizeMode.AutoSize;
            }

            // 
            // Cameo Panel
            // 
            CameoPanel = new System.Windows.Forms.Panel();
            CameoPanel.Location = new System.Drawing.Point(0, 30);
            CameoPanel.Name = "Cameo";
            CameoPanel.Size = new System.Drawing.Size(800, 450);

            //
            // Connect Button
            //

            ConnectButton = new System.Windows.Forms.Button();
            ConnectButton.Location = new System.Drawing.Point(310, 00);
            ConnectButton.Size = new Size(120, 30);
            ConnectButton.Text = "Connect";
            ConnectButton.Click += new EventHandler(ConnectButtonClick);
            ConnectButton.Font = new System.Drawing.Font("Arial", 14F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));

            //
            // Analyze Button
            //

            AnalyzeButton = new System.Windows.Forms.Button();
            AnalyzeButton.Location = new System.Drawing.Point(1600, 30);
            AnalyzeButton.Size = new Size(250, 100);
            AnalyzeButton.Text = "Get face and analyze it";
            AnalyzeButton.Click += new EventHandler(AnalyzeButtonClick);
            AnalyzeButton.Font = new System.Drawing.Font("Arial", 14F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            AnalyzeButton.Enabled = false;

            //
            // Automatic Button
            //

            AutomaticButton = new System.Windows.Forms.Button();
            AutomaticButton.Location = new System.Drawing.Point(1600, 400);
            AutomaticButton.Size = new Size(250, 100);
            AutomaticButton.Text = "Automatic disabled";
            AutomaticButton.Click += new EventHandler(AutomaticButtonClick);
            AutomaticButton.Font = new System.Drawing.Font("Arial", 14F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            AutomaticButton.Enabled = false;


            // 
            // Result label
            //       
            ResultsOfAnalysis = new System.Windows.Forms.Label();
            ResultsOfAnalysis.Location = new System.Drawing.Point(1600, 140);
            ResultsOfAnalysis.Size = new Size(300, 300);
            ResultsOfAnalysis.Text = "Results: \r\n" + " x % " + "happiness\r\n" + " x % " + "sadness\r\n" + " x % " + "neutral\r\n" + " x % " + "surprise\r\n" + " x % " + "anger\r\n" + " x % " + "contempt\r\n" + " x % " + "fear\r\n\r\n";
            ResultsOfAnalysis.Font = new System.Drawing.Font("Arial", 14F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));

            // 
            // MainWindow
            // 

            BackColor = SystemColors.Control;
            ClientSize = new Size(1850, 500);
            Controls.Add(URLLabel);
            Controls.Add(CameraUrl);
            Controls.Add(ConnectButton);
            Controls.Add(CameraDriverLabel);
            Controls.Add(CameraDriverComboBox);
            Controls.Add(SubsciptionKeyLabel);
            Controls.Add(SubsciptionKey);
            Controls.Add(CameoPanel);
            Controls.Add(CameraSnapshot);
            Controls.Add(AnalyzeButton);
            Controls.Add(AutomaticButton);
            Controls.Add(ResultsOfAnalysis);
            Name = "MainWindow";
            Text = "Bosch Microsoft Analytic Sample";
        }

        private void AnalyzeButtonClick(object sender, EventArgs e)
        {
            string PictureFileName = PICTURE_PATH + "\\SnapShot_" + DateTime.Now.ToString("dd_MM_yyyy-HH-mm-ss") + ".jpg";
            string DownloadURL = "http://" + CameraUrl.Text + "/snap.jpg";
            MicrosoftCognitiveServiceRequests.Analyze(PictureFileName, DownloadURL, SubsciptionKey.Text);

        }

        private void AutomaticButtonClick(object sender, EventArgs e)
        {
            if (AutomaticButton.Text.Equals("Automatic disabled"))
            {
                AutomaticButton.Text = "Automatic activated";
                AutomaticActive = true;
                AnalyzeButton.Enabled = false;
            }
            else
            {
                AutomaticButton.Text = "Automatic disabled";
                AutomaticActive = false;
                AnalyzeButton.Enabled = true;
            }

        }

        private void ConnectButtonClick(object sender, EventArgs e)
        {           

            if (ConnectButton.Text.Equals("Connect"))
            {
                CameraConnector.ConnectToCamera(CameraUrl.Text, CameraDriverComboBox.SelectedItem.ToString());
            }
            else
            {
                CameraConnector.Disconnect();
            }
            

        }

        private void MainWindow_FormClosed(object sender, FormClosedEventArgs e)
        {
            CameraConnector.Disconnect();
        }
        public static Bitmap cropThisRect(Bitmap b, Rectangle r)
        {
            return b.Clone(r, b.PixelFormat);
        }
    }
}

File: Example_Clients/CognitiveServiceExample/src/UserInterface.java
import javax.imageio.ImageIO;
import javax.swing.*;
import java.awt.*;
import java.awt.image.BufferedImage;
import java.io.File;


public class UserInterface {

    public static boolean isConnected = false;

    public static String DEFAULT_IP;

    public static String SUBSCRIPTION_KEY_FOR_EMOTION;

    public static String SUBSCRIPTION_KEY_FOR_VISION;

    public static String PICTURE_PATH;

    public static Font FONT =new Font("Arial", Font.CENTER_BASELINE, 14);

    public static JButton analyzeButton;

    private static JButton connectButton;

    public static JButton automaticButton;

    public static boolean automaticActive = false;

    public static JTextField cameraURL;

    private JFrame mainFrame = new JFrame();

    public static JTextField subsciptionKey;

    public static JPanel cameoPanel;

    public static JLabel cameraShotImage;

    public static JLabel faceImage;

    public static JLabel cameraStreaming;

    public static JFormattedTextField resultsOfAnalysis;


    public UserInterface() {
        String ConfigFileName = "Resources\\MSCS.cfg";
        try {
            Configuration.ReadConfig(ConfigFileName);
        }
        catch (Exception ex) {
            ex.printStackTrace();
            System.out.println(("Configfile "
                    + (ConfigFileName + " not found")));
            System.out.println("Please define a configfile named MSCS.cfg to prefill Camera IP and Subscription Key");
        }


    }

    public void InitializeGUIComponents() {

        //
        //  URL Textbox
        //
        cameraURL = new JTextField();
        cameraURL.setText(DEFAULT_IP);
        cameraURL.setSize(150, 20);
        cameraURL.setFont(FONT);
        cameraURL.setLocation(150, 0);
        cameraURL.setName("Camera URL Textbox");

        //
        //  URL Label
        //
        JLabel urlLabel = new JLabel();
        urlLabel.setText("Camera IP : ");
        urlLabel.setSize(150, 20);
        urlLabel.setFont(FONT);
        urlLabel.setLocation(0, 0);
        urlLabel.setName("Camera URL Label");

        //
        //  Subscribtion Key Label
        //
        JLabel subsciptionKeyLabel = new JLabel();
        subsciptionKeyLabel.setText("Subscribtion Key : ");
        subsciptionKeyLabel.setSize(200, 20);
        subsciptionKeyLabel.setFont(FONT);
        subsciptionKeyLabel.setLocation(1250, 0);
        subsciptionKeyLabel.setName("Subscribtion Key Label");
        //
        //  Subscribtion Key Textbox
        //
        subsciptionKey = new JTextField();
        subsciptionKey.setText(SUBSCRIPTION_KEY_FOR_EMOTION);
        subsciptionKey.setSize(400, 20);
        subsciptionKey.setFont(FONT);
        subsciptionKey.setLocation(1450, 0);
        subsciptionKey.setName("Subscribtion Key Textbox");

        //
        //  BoschSmartEdgeDeviceLabel
        //
        JLabel boschSmartEdgeDevice = new JLabel();
        //  Construct an image object from a file in the local directory.
        //  ... This file must exist in the solution.
        String boschSmartEdgeImageFileName = "Resources\\SmartEdgeDevices.png";

        BufferedImage boschSmartEdgeImage = null;

        try {
            boschSmartEdgeImage = ImageIO.read(new File(boschSmartEdgeImageFileName));
        } catch (Exception ex) {
            ex.printStackTrace();
        }

        if ((boschSmartEdgeImage != null)) {
            ImageIcon icon = new ImageIcon(boschSmartEdgeImage);
            boschSmartEdgeDevice.setIcon(icon);
            boschSmartEdgeDevice.setName("CameraSnapshotBox");
        }

        //
        //  CameraStreamingBox
        //
        cameraStreaming = new JLabel();

        //
        //  CameraStreamingBox
        //
        cameraShotImage = new JLabel();

        //
        //  CameraStreamingBox
        //
        faceImage = new JLabel();


        //
        //  Connect Button
        //
        connectButton = new JButton();
        connectButton.setLocation(310, 0);
        connectButton.setSize(120, 30);
        connectButton.setText("Connect");
        connectButton.setFont(FONT);
        ConnectButtonListener connectButtonListener = new ConnectButtonListener();
        connectButton.addActionListener(connectButtonListener);

        //
        //  Analyze Button
        //
        analyzeButton = new JButton();
        analyzeButton.setLocation(1600, 30);
        analyzeButton.setSize(250, 100);
        analyzeButton.setText("Get face and analyze it");
        analyzeButton.setFont(FONT);
        analyzeButton.setEnabled(true);
        AnalyzeButtonListener analyzeButtonListener = new AnalyzeButtonListener();
        analyzeButton.addActionListener(analyzeButtonListener);

        //
        //  Automatic Button
        //
        automaticButton = new JButton();
        automaticButton.setLocation(1600, 400);
        automaticButton.setSize(250, 100);
        automaticButton.setText("Automatic disabled");
        automaticButton.setFont(FONT);
        automaticButton.setEnabled(false);
        AutomaticButtonListener automaticButtonListener = new AutomaticButtonListener();
        automaticButton.addActionListener(automaticButtonListener);

        //
        //  Result label
        //
        resultsOfAnalysis = new JFormattedTextField("");
        resultsOfAnalysis.setLocation(1600, 140);
        resultsOfAnalysis.setSize(300, 300);
        resultsOfAnalysis.setText("Results: \r\n" + (" x % " + ("happiness\r\n" + (" x % " + ("sadness\r\n" + (" x % " + ("neutral\r\n" + (" x % " + ("surprise\r\n" + (" x % " + ("anger\r\n" + (" x % " + ("contempt\r\n" + (" x % " + "fear\r\n\r\n"))))))))))))));
        resultsOfAnalysis.setFont(FONT);

        //
        //  MainWindow
        //

        GridBagLayout gridbag = new GridBagLayout();
        GridBagConstraints gridBagConstraints = new GridBagConstraints();

        mainFrame.setLayout(new GridLayout(4, 3));
        mainFrame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        mainFrame.setSize(1200, 800);



        mainFrame.add(urlLabel);
        mainFrame.add(boschSmartEdgeDevice);
        mainFrame.add(subsciptionKeyLabel);

        mainFrame.add(cameraURL);
        mainFrame.add(connectButton);
        mainFrame.add(subsciptionKey);

        mainFrame.add(cameraShotImage);
        mainFrame.add(cameraStreaming);
        mainFrame.add(faceImage);

        mainFrame.add(automaticButton);
        mainFrame.add(analyzeButton);
        mainFrame.add(resultsOfAnalysis);

        mainFrame.setName("Bosch Microsoft Analytic Sample");



        //Display the window.
        mainFrame.pack();
        mainFrame.setVisible(true);

    }




    public static BufferedImage cropImage(BufferedImage bufferedImage, Rectangle rectangle) {
        return bufferedImage.getSubimage(rectangle.x, rectangle.y, rectangle.width, rectangle.height);
    }

    public static boolean isConnected() {
        return isConnected;
    }

    public static void changeConnectionStatus()
    {
        if(isConnected)
        {
            isConnected = false;
            connectButton.setText("Connect");
        }
        else {
            isConnected = true;
            connectButton.setText("Disconnect");
        }
    }

}
